src/main/java/persistence/sql/column/JoinTableColumns.java

```diff
@@ -0,0 +1,54 @@
+package persistence.sql.column;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class JoinTableColumns {
+
+    private static final String COMMA = ", ";
+    private final List<JoinTableColumn> values;
+
+    public JoinTableColumns(List<JoinTableColumn> values) {
+        this.values = new ArrayList<>(values);
+    }
```

Comment:
```suggestion
    private final List<JoinTableColumn> values;
    private final boolean eager;
    private final boolean lazy;

    public JoinTableColumns(List<JoinTableColumn> values) {
        this.values = new ArrayList<>(values);
        this.eager = values.stream()
                .anyMatch(joinTable -> !joinTable.getAssociationEntity().isLazy());
        this.lazy = values.stream()
                .anyMatch(joinTable -> joinTable.getAssociationEntity().isLazy());
    }
    public boolean hasEager() {
        return eager;
    }

    public boolean hasLazy() {
        return lazy;
    }
```
취향 차이인데 어차피 일급컬렉션을 생성하셨으니 이쪽에 로직을 좀 두어도 괜찮아 보입니다 :)

Reply:
좋습니다! 반영했습니다~!